### ISSUE TITLE
[1.10.2] Galactic Craft 4 - java.lang.ArrayIndexOutOfBoundsException: 7

### ISSUE THREAD

**Issue by llloyd4** [micdoodle8/Galacticraft#2868]
Ok, so. I peeked at the troubleshooting guide. My first 4 lines isn't even remotely the same as the troubleshooting guide so I peeked in here. Searched out ' java.lang.ArrayIndexOutOfBoundsException: 7 ' with no results, so hopefully I fulfilled all the requirements. 

Minecraft: 1.10.2
Forge: 1.10.2-12.18.3.2316
Galacticraft: 1.10.2-4.0.0.109

The first four lines of the crash report is 

java.lang.ArrayIndexOutOfBoundsException: 7
	at micdoodle8.mods.galacticraft.core.items.ItemTier1Rocket.func_77624_a(ItemTier1Rocket.java:168)
	at net.minecraft.item.ItemStack.func_82840_a(ItemStack.java:615)
	at com.raoulvdberge.refinedstorage.gui.grid.stack.GridStackItem.getTooltip(GridStackItem.java:64)
	at com.raoulvdberge.refinedstorage.gui.grid.GuiGrid.drawForeground(GuiGrid.java:376)

Full crash log : https://pastebin.com/Bc1usRNh
Due to the mention of Refined Storage, same issue on the Refined Storage Github - https://github.com/raoulvdberge/refinedstorage/issues/1335 - And pretty quickly Raoul, mod author for RS, ruled it a bug in your mod and closed the RS Github report. So, that's a sort of a relief, at least I can focus here on a fix to make my unplayable world playable.

**Comment by radfast** [comment]
It's not a bug in our mod!  It's something wrong with the item.

Refined Storage is attempting to draw a Tier 1 Rocket item with an item damage value of 7. 

That's a damage value which doesn't exist for Galacticraft rockets.  There is no way that Galacticraft can make that item, and you won't find it in JEI for example.  For rockets, the allowed damage values are 0 through 4 only.

If you spawned the rocket in using server commands or a command block, then I guess you could get a rocket with an illegal damage value but you shouldn't do that.

The only other ways this can happen would be (1) a corrupted save (or if you used some kind of editor to modify your save) or (2) if another mod is creating an illegal item somehow.  You might want to look again at Refined Storage and any other mod which your items are passing through - for example if you have item pipes or similar - to see if somehow they can have corrupted the data on an item.  Doesn't seem very likely as it would be a bug affecting items from many mods, but I can't come up with any other explanation for how you have this illegal item.

Anyhow, 99% sure it's not a bug in Galacticraft so I'm closing this an "issue".  Even though this is closed, you're welcome to ask questions or to discuss anything.  If something shows it is a bug in Galacticraft we can re-open this issue report.

**Comment by llloyd4** [comment]
Which is odd since I am only opening the Crafting Grid to craft and get things out of my storage and it only started when I installed Galacticraft.

**Comment by radfast** [comment]
Like I already said, the problem here is you have an impossible item - in just the same way as a vanilla wood log with a damage value of 25 is an impossible item (there are only 7 types of vanilla wood).

How did you make the Tier 1 Rocket item?  Did you craft it in the NASA workbench using the Tier 1 Rocket recipe or did you make it some other way?  How did you place it in the Refined Storage Crafting Grid?  Please provide any possible explanation of how this item comes to have the "impossible" damage value of 7 in your game?

Maybe also start a fresh world to test with your modpack, in Creative Mode, and test the Crafting Grid with a Galacticraft rocket item in that new world to see if the same thing happens.

**Comment by llloyd4** [comment]
That's what I am saying, I hadn't crafted anything yet, from the moment I
put in GC4 it has crashed and not let me craft anything. I did, I *think*,
sort it out. I backed up my world, uninstalled GC4, reloaded my world,
clicked Yes to the missing items, went though everything in the RS network,
items in storage, patterns, etc and made sure there were no lingering
invalid patterns, etc and then shut down, reinstalled GC4, and it seems
stable now. Only problem so far is I am not finding any meteorites and the
/summon command keeps coming up as "Cannot summon" even if I copy / paste
it from the website. I've tried it both on the moon and on the Overworld.

**Comment by radfast** [comment]
OK, good, sounds like you sorted it out.

You're not really supposed to swap in and out mods in an existing world, that can produce issues.

The meteorites, if you spend some time on the Moon - for example constructing a Moon base - then a meteorite should spawn near you on average once every 3-4 minutes (slower if the server has a very slow tps).

The summon command is different in different Minecraft versions, I've updated the wiki with the correct commands now: https://wiki.micdoodle8.com/wiki/Fallen_Meteor#Obtaining

**Comment by llloyd4** [comment]
Ah, ok, and the wiki says the Circuit Maker is suppose to be automatable.
But I put a hopper feeding into a chest under it and it's not pulling out
of the output slot. >.< And, yeah, my world is old, I am looking for a new
challenge. I've used this in version 3 and got as far as Mars but this was
before Venus and the Asteroid Belt got added. But, yeah, that's on the next
To Do that I am setting up for, a moon base. Just trying to automate the
machines a bit to autocraft the components I'll need.
